pkg/jobs/update.go

```diff
@@ -171,6 +171,17 @@ WHERE id = $1
 	if err := updateFn(u.txn, md, &ju); err != nil {
 		return err
 	}
+	if ju.md.Status != "" {
```

Comment:
`update` is not just called during a status change, and further, md.Status will always be not nil. So, this current change will log structured events on many more events than a status change. I don't think this is intended?

Reply:
I think your original call sites were fine, and calling this in a commit trigger is a nice way to avoid the retry problem!

Reply:
One reason why i wanted to move it here is that the update function is guaranteed to be in a transaction which made it safe to use the commit trigger and we know that this function is always going to be invoked for status changes. 

From reading the code, `ju.md.Status` is only set on status updates, which i saw by checking usages of `JobUpdater.UpdateStatus`, but i guess its not good rely on this always being true in the future. What if we leave it here and change the condition to something like: 
`if ju.md.Status != "" && (status != ju.md.Status || (status == StatusRunning && ju.md.Status == StatusRunning)) {...}`

Reply:
huh, I see. I'm coming around to this call site! While your argument does hold for all status changes that occur via `UpdateStatus()`, sadly we do not call `UpdateStatus(PAUSED)` or `UpdateStatus(REVERTING)` (for all paths) and instead manually change the status via this query. https://github.com/msbutler/cockroach/blob/butler-skip-auto-replan-stress/pkg/jobs/adopt.go#L530

I think if you added a call `maybeLogStateChangeStructured` around this query and added a little dosctring explaining the logic in the `update()` path, I'd feel good about this approach. @dt should also take a look at this.

Reply:
Awesome, thanks for the feedback! I updated the conditional here, added a doc string, and added the log to `servePauseAndCancelRequests` as well.

Reply:
Thanks for addressing this! I left a few more comments, but will not have the time to give this a full review before heading on PTO tomorrow. I'd ping @stevendanna or @dt for a final review.